### PR TITLE
Feature/statistics

### PR DESCRIPTION
**Pull request by lestarcdog**
I made the HTTPUtils a Spring bean and modified the two binding classes to inject it into them..

In the previous commits I also refactored the Request, Response, OutputResponse events classes to have a base class which contains the [event group id](https://github.com/52North/SOS/issues/216).

### CODE REVIEW

File: src/main/java/org/n52/iceland/binding/SimpleBinding.java [hunk 1]
```diff
@@ -74,8 +73,18 @@
...
     private EncoderRepository encoderRepository;
     private DecoderRepository decoderRepository;
+    private HTTPUtils httpUtils;
+    
+    public HTTPUtils getHttpUtils() {
+		return httpUtils;
+	}
 
     @Inject
+	public void setHttpUtils(HTTPUtils httpUtils) {
```
Comment (autermann) [inline src/main/java/org/n52/iceland/binding/SimpleBinding.java]: Could you please switch the indentation to 4 spaces?

File: src/main/java/org/n52/iceland/util/http/HTTPUtils.java [hunk 2]
```diff
@@ -46,180 +53,200 @@
...
 
 		@Override
 		public MediaType getEncodedContentType() {
 			return response.getContentType();
 		}
-    }
+	}
 
-    public interface Writable {
-        void write(OutputStream out, ResponseProxy responseProxy) throws IOException;
```
Comment (autermann) [inline src/main/java/org/n52/iceland/util/http/HTTPUtils.java]: Same here...

File: src/main/java/org/n52/iceland/util/http/HTTPUtils.java [hunk 3]
```diff
@@ -46,180 +53,200 @@
...
-                responseProxy.setContentLength(response.getContentLength());
-            }
-            response.writeToOutputStream(out);
-        }
-
-        @Override
-        public boolean supportsGZip() {
-            return response.supportsGZip();
-        }
+	private static final Logger LOGGER = LoggerFactory.getLogger(HTTPUtils.class);
```
Comment (autermann) [inline src/main/java/org/n52/iceland/util/http/HTTPUtils.java]: And here...

**Comment by lestarcdog** [comment]
Now it clears some of the whitespaces also.

**Comment by autermann** [comment]
So much better, thanks :)

**Comment by nuest** [comment]
Thanks for merging this - one question though @autermann : Do you see any potential issues with using the thread's id to connect the request and response events together?

**Comment by nuest** [comment]
this actually requires a setting definition for ``statistics.counting-outputstream`` which is not part of iceland yet.

@lestarcdog for this it would be really useful to split up your code into generic (= iceland) and spefic (= SOS) code so that such a setting can be included in iceland (default value respectively documentation and examples setttings beans)

**Pull request merged**